src/scala/com/github/johnynek/bazel_deps/IO.scala

```diff
@@ -115,83 +111,194 @@ object IO {
+        Try {
+          // get the java path
+          val file = fileFor(f)
+          // require(file.isDirectory, s"$f is not a directory")
+          val path = file.toPath
+          if (!removeHidden) {
             Files.walkFileTree(
               path,
               new SimpleFileVisitor[JPath] {
                 override def visitFile(
                     file: JPath,
                     attrs: BasicFileAttributes
                 ) = {
-                  Files.delete(file)
-                  FileVisitResult.CONTINUE
-                }
-                override def postVisitDirectory(
-                    dir: JPath,
-                    exc: IOException
-                ) = {
-                  Files.delete(dir)
+                  if (file.getFileName.startsWith(".") && !removeHidden) { // Hidden!
+                    throw new Exception(
+                      s"Encountered hidden file ${file.getFileName}, and should not remove hidden files/folders. Aborting."
+                    )
+                  }
                   FileVisitResult.CONTINUE
                 }
               }
             )
-            ()
           }
-        case WriteFile(f, d) =>
-          Try {
-            val os = new FileOutputStream(fileFor(f))
-            try os.write(d.value.getBytes(charset))
-            finally { os.close() }
+
+          Files.walkFileTree(
+            path,
+            new SimpleFileVisitor[JPath] {
+              override def visitFile(
+                  file: JPath,
+                  attrs: BasicFileAttributes
+              ) = {
+                Files.delete(file)
+                FileVisitResult.CONTINUE
+              }
+              override def postVisitDirectory(
+                  dir: JPath,
+                  exc: IOException
+              ) = {
+                Files.delete(dir)
+                FileVisitResult.CONTINUE
+              }
+            }
+          )
+          ()
+        }
+      case WriteFile(f, d) =>
+        Try {
+          val os = new FileOutputStream(fileFor(f))
+          try os.write(d.value.getBytes(charset))
+          finally { os.close() }
+        }
+      case WriteGzipFile(f, d) =>
+        Try {
+          import java.util.zip.GZIPOutputStream;
+          val os = new GZIPOutputStream(new FileOutputStream(fileFor(f)))
+          try os.write(d.value.getBytes(charset))
+          finally { os.close() }
+        }
+      case _ => super.apply(o)
+    }
+  }
+
+  sealed abstract class CheckException(val message: String) extends Exception(message) {
+    def path: IO.Path
+  }
+
+  object CheckException {
+    case class DirectoryMissing(path: IO.Path) extends CheckException(s"directory ${path.asString} expected but missing")
+    case class WriteMismatch(path: IO.Path, current: Option[String], expected: String, compressed: Boolean) extends CheckException(
+      (if (compressed) "compressed " else "") + s"file ${path.asString} didn't match."
```

Comment:
You don't do anything with `current` and `expected` here, but I think you should! 😄 Particularly on the case where `current` is None, I think you should probably have the error return something like `file {path} does not exist!`